wrangler-core/src/main/java/io/cdap/directives/column/CreateRecord.java

```diff
@@ -101,4 +106,20 @@ public Mutation lineage() {
       .relation(Many.columns(columns), targetColumn)
       .build();
   }
+
+  @Override
+  public Relation transform(RelationalTranformContext relationalTranformContext,
+                            Relation relation) {
+    java.util.Optional<ExpressionFactory<String>> expressionFactory = SqlExpressionGenerator
+            .getExpressionFactory(relationalTranformContext);
+    if (!expressionFactory.isPresent()) {
+      return new InvalidRelation("Cannot find an Expression Factory");
+    }
+    return relation.setColumn(targetColumn, expressionFactory.get().compile(String
+            .format("struct(%s)", getColumnString(columns))));
+  }
+
+  public String getColumnString(String[] columns) {
```

Comment:
Can just have a string variable instead of a function to make it easier to read. Similarly for WriteAsJsonObject (and other places where just String.join is used)